k8s/core/configmap/configmap_lock_v2_test.go

```diff
@@ -402,3 +402,72 @@ func TestCMLockTTL(t *testing.T) {
 	require.Equal(t, owner, lockOwners[key])
 	require.Equal(t, currTime.Add(testLockTTL), lockExpirations[key])
 }
+
+func TestIsKeyLocked(t *testing.T) {
+	setUpConfigMapTestCluster(t)
+	cmIntf, err := New("px-configmaps-lock-is-key-locked-test", nil, testLockTimeout, testLockAttempts, testLockRefreshDuration, testLockTTL)
+	require.NoError(t, err, "Unexpected error on New")
+	cm := cmIntf.(*configMap)
+
+	id1 := "is-key-locked-id1"
+	id2 := "is-key-locked-id2"
+	key := "is-key-locked-key"
+
+	locked, _, err := cm.IsKeyLocked(key, id1)
+	require.NoError(t, err)
+	require.False(t, locked)
+
+	// lock the key
+	err = cm.LockWithKey(id1, key)
+	require.NoError(t, err)
+	require.True(t, cm.kLocksV2[key].refreshing)
+
+	// check if locked and owner
+	locked, owner, err := cm.IsKeyLocked(key, id1)
+	require.NoError(t, err)
+	require.True(t, locked)
+	require.Equal(t, id1, owner)
+	locked, owner, err = cm.IsKeyLocked(key, id2)
+	require.NoError(t, err)
+	require.True(t, locked)
+	require.Equal(t, id1, owner)
+
+	// stop goroutine and make lock expired
+	close(cm.kLocksV2[key].done)
+	cm.kLocksV2[key].unlocked = true
+	time.Sleep(2 * testLockTTL)
```

Comment:
You don't need to sleep here. You can patch time and fast forward to what time you need.
you can keep a variable patchedTime and make the patched function return it. Before the next step, you can assign to patchedTime what value you expect. And it can be done multiple times in the same test.